core/src/main/java/org/jdbi/v3/core/Jdbi.java

```diff
@@ -362,7 +363,16 @@ public Handle open() {
             }
             LOG.trace("Jdbi [{}] obtain handle [{}] in {}ms", this, h, MILLISECONDS.convert(stop - start, NANOSECONDS));
             return h;
-        } catch (SQLException e) {
+        } catch (Exception e) {
```

Comment:
this makes me super-uncomfortable. There is a single bug report about a problem with connection closing and this is a change at one of the most fundamental pieces of Jdbi, affecting the most fundamental path in the whole code base which is probably executed a billion times a day all over the java ecosystem without any problems.

Why touch this and if touching it, there needs to be a bunch of tests that *prove* that this low level code (this replaces one line of straightforward catch and rethrow with a bunch of manual resource manipulations) actually is an improvement. I don't believe it is.

Reply:
I spend a lot of time and wrote a lot of tests in the 3.35.0 cycle to prove that the new code not only behaves better but does not leak resources anymore. I don't think introducing this manual piece-by-piece resource releasing and bypassing the mechanism that is used everywhere else in post-3.35.0 code is the right thing to do.

Reply:
Hi @hgschmie 
I totally understand your concerns. 

> There is a single bug report about a problem with connection closing

This is true, it is a single bug report and it was originally about connection closing. However yesterday I have updated the bug report because it turns out that the problem that I found is more generic and case when db connection was closed can be one of the "implementation" of this issue.

The pool has basically two sets of methods, 
one for acquiring the object and second one for returning/destroying the object when it is no longer needed. Both methods are called by pool client which in this case is JDBI. It is the pool client who knows when to return the object back to the pool, so its up to the caller (JDBI i this case) to make sure that object is always return back to the pool right? At lest to make sure that proper API is called. Forgive me but I can not say that this is a pool bug, but maybe I'm missing something.

The `org.jdbi.v3.core.Jdbi::open()` gets object from the pool. If there will be an exception thrown anywhere in `org.jdbi.v3.core.Jdbi::open()` method, after the connection was acquired from pool, the connection object will not be returned back to the pool.

I would say that `org.jdbi.v3.core.Jdbi::open()` should have exception handling logic which will guaranty that connection object will be properly returned to the pool in case of exception. Currently it is not the case.

**The specific use cases for connection close:**

1. The connection is closed from the client side, as described in this [bug report](https://github.com/jdbi/jdbi/issues/2446).
In this case it turns out that exception will be thrown by this [line](https://github.com/jdbi/jdbi/blob/299268ba7b00a2845e05f7942fb416147d603f18/core/src/main/java/org/jdbi/v3/core/Handle.java#L115C17-L115C17) in Handle's constructor.
this.transactionHandler = transactionHandler.specialize(this); so even before registering any Cleanable objects
In this case, since exception was thrown from Handle constructor, there is no Handle instance to close/clean at the end.

2. The second use case is where the connection is closed from server side, for example Data Base restart.
In our system we are using Jdbi's PostgresPlugin. In org.jdbi.v3.core.Jdbi::open() method, after Handle object is created there is this call:
```
for (JdbiPlugin p : plugins) {
        h = p.customizeHandle(h);
}
```
This will throw an exception for PostgresPlugin.
In result the open() method will not return any Handle reference so no Handle will be closed/clean up hence connection object will not be returned to the pool. The stack trace for this case -
[JdbiTrace.txt](https://github.com/jdbi/jdbi/files/12276766/JdbiTrace.txt)

To reproduce this, please pull my repository (I added PostgresPlugin) and simply restart Postgresql container on a break point just after where connection object is acquired from the pool.

However this problem will happen in case of any exception thrown in Jdbi::open() method, connection issue is only one of them.

Reply:
I added test cases to show this behavior in a unit test that we can commit to the repo.